Assets/PlayroomKit/PlayroomKit.cs

```diff
@@ -971,6 +971,167 @@ public class Dpad
+        }
+
+        // Default Mode
+        public static void RpcCall(string name, object data, Action callbackOnResponse)
+        {
+            RpcCall(name, data, RPCModes.ALL, callbackOnResponse);
+        }
+
+        [MonoPInvokeCallback(typeof(Action))]
+        private static void InvokeOnResponseCallback()
+        {
+            foreach (var name in RpcEventNames)
+            {
+                try
+                {
+                    if (OnResponseCallbacks.TryGetValue(name, out List<Action> callbacks))
+                    {
+                        foreach (var callback in callbacks)
+                        {
+                            callback?.Invoke();
+                        }
+
+                        RpcEventNames.Remove(name);
+                        OnResponseCallbacks.Remove(name);
+                    }
+                }
+                catch (Exception ex)
+                {
+                    Debug.LogError($"C#: Error in Invoking callback for RPC event name: '{name}': {ex.Message}");
+                }
+            }
+        }
+
+
+        private static string ConvertToJson(object data)
+        {
+            if (data == null)
+            {
+                return null;
+            }
+            else if (data.GetType().IsPrimitive || data is string)
+            {
+                return data.ToString();
+            }
+            if (data is Vector2 vector2)
+            {
+                return JsonUtility.ToJson(vector2);
+            }
+            else if (data is Vector3 vector3)
+            {
+                return JsonUtility.ToJson(vector3);
+            }
+            else if (data is Vector4 vector4)
+            {
+                return JsonUtility.ToJson(vector4);
+            }
+            else if (data is Quaternion quaternion)
+            {
+                return JsonUtility.ToJson(quaternion);
+            }
+            else
+            {
+                return ConvertComplexToJson(data);
+            }
+        }
+
+        private static string ConvertComplexToJson(object data)
+        {
+            if (data is IDictionary dictionary)
+            {
+                JSONObject dictNode = new JSONObject();
+                foreach (DictionaryEntry entry in dictionary)
+                {
+                    dictNode[entry.Key.ToString()] = ConvertToJson(entry.Value);
+                }
+                return dictNode.ToString();
+            }
+            else if (data is IEnumerable enumerable)
+            {
+                JSONArray arrayNode = new JSONArray();
+                foreach (object element in enumerable)
+                {
+                    arrayNode.Add(ConvertToJson(element));
+                }
+                return arrayNode.ToString();
+            }
+            else
+            {
+                Debug.Log($"{data} is '{data.GetType()}' which is currently not supported by RPC!");
```

Comment:
Can you drop an explainer why this is necessary? Why can't we support these types?